src/Entity/ListBuilder/EdgeEntityListBuilder.php

```diff
@@ -126,10 +126,20 @@ protected function buildEntityIdQuery(): QueryInterface {
   public function render() {
     $settings = $this->getDisplaySettings();
     if ($this->usingDisplayType(static::VIEW_MODE_DISPLAY_TYPE)) {
-      return $this->renderUsingViewMode($settings['view_mode']);
+      $build = $this->renderUsingViewMode($settings['view_mode']);
     }
+    else {
+      $build = parent::render();
+    }
+
+    // Add cache contexts.
+    $build['#cache'] = [
+      'contexts' => $this->entityType->getListCacheContexts(),
+      'tags' => $this->entityType->getListCacheTags(),
```

Comment:
The entity type list cache tags and contexts are not user aware, which means the render array will get cached the same for all of the users. We need to add the `user` and maybe also `user.permissions` to  the contexts to make the render array aware that the response changes per user.

Reply:
Added. Thanks.